scripts/toolchain/install-toolchain.sh

```diff
@@ -0,0 +1,229 @@
+
+declare install_all with_yarn
+install_all="true"
+with_yarn="false"
+
+while (( "$#" )); do
+  case "$1" in
+    -h|--help)
+      # return early with help info when requested
+      usage
+      exit 0
+      ;;
+    --runtime-only)
+      install_all="false"
+      shift
+      ;;
+    --with-yarn)
+      with_yarn="true"
+      shift
+      ;;
+    -*|--*=)
+      usage
+      exit 1
+      ;;
+    *)
+      shift
+      ;;
+  esac
+done
+
+if [[ "${PATH}" =~ "/usr/local/bin" ]]; then
+    echo "Cannot install utilities. \"/usr/local/bin\" is not part of home-path."
+fi
+
+if ! [ -w "/usr/local/bin" ]; then
+    echo "Cannot install utilities. \"/usr/local/bin\" is not writable."
+fi
+
+if ! [ -d "/opt" ] && ! [ -w "/opt" ] || ! [ -w "/" ]; then
+    echo "Cannot install utilities. \"/opt\" does not exist or is not writable."
+fi
+
+declare download_dir="/tmp/hopr-toolchain/download"
+mkdir -p ${download_dir}
+
+function install_rustup() {
+    if ! command -v rustup; then
+        echo "Installing Rustup"
+        # Get rustup but install toolchain later
+        curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- ----no-modify-path --default-toolchain none -y
+        # Set PATH such that `cargo` is available within this script
+        PATH=${PATH}:${HOME}/.cargo/bin
+    fi
+}
+
+function install_cargo() {
+    if ! command -v cargo; then
+        local rust_toolchain_toml_path="${mydir}/../../rust-toolchain.toml"
+        local rust_version=$(sed -En 's/^channel = "([0-9a-z.]*)"$/\1/p' ${rust_toolchain_toml_path})
+        echo "Installing Cargo, Rust compiler v${rust_version}"
+        local target=$(sed -En 's/^targets = \[ "([a-z0-9-]*)" \]$/\1/p' ${rust_toolchain_toml_path})
+        local profile=$(sed -En 's/^profile = "([a-z]*)"$/\1/p' ${rust_toolchain_toml_path})
+
+        # Always install the version specified in `rust-toolchain.toml`
+        $HOME/.cargo/bin/rustup toolchain install ${rust_version} --profile ${profile} --target ${target}
+        source "$HOME/.cargo/env"
+    fi
+}
+
+function install_wasm_pack() {
+    if ! command -v wasm-pack; then
+        cd ${download_dir}
+        local rust_cargo_toml_path="${mydir}/../../Cargo.toml"
+        local wasm_pack_release=$(sed -En 's/^wasm-pack = \"([0-9.]*)\"$/v\1/p' ${rust_cargo_toml_path})
+        echo "Installing wasm-pack ${wasm_pack_release}"
+        local ostype="$(uname -s)"
+        local cputype="$(uname -m)"
+        case "${ostype}" in
+            Linux | linux)
+                ostype="unknown-linux-musl"
+                ;;
+            Darwin)
+                ostype="apple-darwin"
+                ;;
+            *)
+                echo "no precompiled binaries available for OS: ${ostype}"
+            ;;
+        esac 
+        curl -fsSLO --compressed "https://github.com/rustwasm/wasm-pack/releases/download/${wasm_pack_release}/wasm-pack-${wasm_pack_release}-${cputype}-${ostype}.tar.gz"
```

Comment:
Also here, please use the version from `Cargo.lock` instead of downloading latest.

Reply:
see line 112 ;-)